### PR TITLE
Fix Sentry configuration for multiprocessing environment

### LINKED ISSUE #2399: Ingestion server errors not sent to Sentry
## Description

Recently an error in the ingestion server's `TableIndexer::create_and_populate_filtered_index` method was not caught and set to Sentry. We only found the error as a result of the DAG triggering the method catching it.

Originally caught by @krysal and @AetherUnbound 

## Reproduction

I'm not sure how to reproduce this and haven't tried yet. I would start by configuring Sentry locally and causing an error in a `TableIndexer` method. I suspect the issue comes down to not having configured Sentry to work in a multiprocessing environment. Docs for that are here: https://docs.sentry.io/platforms/python/troubleshooting/#addressing-concurrency-issues

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Fixes

Fixes #2399 by @sarayourfriend 

## Description

Relevant Sentry docs:

- Multi-thread Sentry: https://docs.sentry.io/platforms/python/troubleshooting/#addressing-concurrency-issues
- Draining the client: https://docs.sentry.io/platforms/python/configuration/draining/?original_referrer=https%3A%2F%2Fdocs.sentry.io%2Fplatforms%2Fpython%2Ftroubleshooting%2F

This PR adds a wrapper function around our multiprocessing entrypoint that runs the worker's actual task wrapped inside a Sentry context. This allows the client cloned by the scheduler (in api.py) to capture messages and exceptions. We also catch any exceptions that occur and pass them to the client. Before allowing the worker to exit, the wrapper flushes the client so that all messages actually make their way to Sentry instead of being lost when the thread stops.

## Testing Instructions

The best way to test this is to cause an exception locally and confirm that it sends to Sentry.

To do that, configure `SENTRY_DSN` in `ingestion_server/.env`. Please also set `ENVIRONMENT=local` to prevent alerts showing up in our production monitoring channels.

Then add something like `raise Exception("Testing sentry ingestion server")` in `process_task` and confirm that the exception is captured. You can also try using `sentry_sdk.capture_message` etc to confirm those are also sent. Check in the Sentry project for the DSN you used to confirm everything appears there.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like`Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [N/A] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.
- [N/A] I ran the DAG documentation generator (if applicable).

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: ingestion_server/ingestion_server/tasks.py [hunk 1]
```diff
@@ -153,6 +156,47 @@ def get_task_status(self, task_id) -> dict:
...
+
+    We cannot use the convenient ``ThreadingIntegration`` because it
+    does not support ``multiprocessing``, only the legacy ``threading``
+    module (we use ``multiprocessing``)
+    """
+
+    @wraps(fn)
+    def fn_with_sentry(*args, **kwargs):
+        hub = kwargs.pop("sentry_hub")
+        with hub:
```
Comment (dhruvkb) [inline ingestion_server/ingestion_server/tasks.py:176]: Nice!

**Review by dhruvkb — APPROVED** [review]
LGTM, thanks for the clear testing instructions. I saw your [test event](https://openverse.sentry.io/share/issue/7ff227759a2442b8a273ad981eac40ea/) when triaging Sentry for the ingestion server.

**Review by krysal — APPROVED** [review]
Looks great! The exception was captured ✅

CleanShot 2023-07-04 at 13 55 19@2x

**Review by stacimc — APPROVED** [review]
Screen Shot 2023-07-04 at 11 08 12 AM

LGTM :) I tried raising an Exception within `ingest_upstream` within `perform_task` and got a second event as well (id eb5c2fc9cf2d4f24a03436c2d87a908e).

**Pull request merged**